lib/matplotlib/tests/test_mathtext.py

```diff
@@ -134,7 +134,7 @@
     r'$\sum x\quad\sum^nx\quad\sum_nx\quad\sum_n^nx\quad\prod x\quad\prod^nx\quad\prod_nx\quad\prod_n^nx$',  # GitHub issue 18085
     r'$1.$ $2.$ $19680801.$ $a.$ $b.$ $mpl.$',
     r'$\text{text}_{\text{sub}}^{\text{sup}} + \text{\$foo\$} + \frac{\text{num}}{\mathbf{\text{den}}}\text{with space, curly brackets \{\}, and dash -}$',
-
+    r'$\sum_{\substack{k = 1}{k \neq \lfloor n/2\rfloor}}^{n}P(i,j) \sum_{\substack{i \neq 0}{-1 \leq i \leq 3}{1 \leq j \leq 5}} F^i(x,y)$',
```

Comment:
Although I believe that your syntax makes very much sense, it appears as if the existing `\substack` command is simply using `\\` for next line.

```suggestion
    r'$\sum_{\substack{k = 1\\ k \neq \lfloor n/2\rfloor}}^{n}P(i,j) \sum_{\substack{i \neq 0\\ -1 \leq i \leq 3\\ 1 \leq j \leq 5}} F^i(x,y)$',
```

Apart from that, the outcome looks good! Maybe one would like to test with `\left \lfloor \frac{n}{2} \right\rfloor` as well to see how that behaves?

Reply:
Not really sure how to go about adding the `\\` as next-line behavior to the `substack` input due to the python's handling of escape characters, I'm currently still looking into it.

Added the `\left \lfloor \frac{n}{2} \right\rfloor` as a test with the current implemention.

Reply:
Maybe the parser expert @anntzer has some input?

Reply:
I would expect pyparsing's [`DelimitedList`](https://pyparsing-docs.readthedocs.io/en/latest/pyparsing.html#pyparsing.DelimitedList) would be helpful here

Note, however, that it looks like it is in a slightly awkward transition from the `delimited_list` function to the `DelimitedList` class, the former being deprecated upon the introduction of the latter in pyparsing 3.1.0 (which we are incompatible with for other reasons right now, but should fix)

As such may need something like:

```python
if pyparsing.__version__info__ < (3, 1):
    from pyparsing import delimited_list as DelimitedList
else:
    from pyparsing import DelimitedList
```